Snake.js

```diff
@@ -2,31 +2,86 @@ class Snake {
     constructor(headX, headY) {
         let _head = new SnakeSegment(headX, headY, null);
 
-        this.getHead = () => {
-            return _head;
-
-        };
-
-        this.move = (direction) => {
+        let _getMoveDiffVector = (direction) => {
+            var diffVector = {
+                x: 0,
+                y: 0
+            };
             if (direction === Directions.Right) {
-                //TODO
-                _head = new SnakeSegment(_head.getX() + 1, _head.getY(), _head.getNext());
+                diffVector = {
+                    x: 1,
+                    y: 0
+                };
             }
             if (direction === Directions.Left) {
-                _head = new SnakeSegment(_head.getX() - 1, _head.getY(), _head.getNext());
+                diffVector = {
+                    x: -1,
+                    y: 0
+                };
             }
             if (direction === Directions.Up) {
-                _head = new SnakeSegment(_head.getX(), _head.getY() - 1, _head.getNext());
+                diffVector = {
+                    x: 0,
+                    y: -1
+                };
             }
             if (direction === Directions.Down) {
-                _head = new SnakeSegment(_head.getX(), _head.getY() + 1, _head.getNext());
+                diffVector = {
+                    x: 0,
+                    y: 1
+                };
             }
+            return diffVector;
+        }
+
+        this.getHead = () => {
+            return _head;
+        };
+
+        this.move = (direction) => {
+            let diffVector = _getMoveDiffVector(direction);
```

Comment:
całe poruszanie jest teraz zrobione na zasadzie:
1 dodaj jako głowe nową pozycję zgodnie z aktualnym kierunkiem (to doda ci jeden segment do snejka)
2 usuń ostatni segment - to zapewni że po wykonaniu "move" mamy cały czas tą samą ilość segmentów (move nie dodaje segmentu tylko je porusza (dodawanie segmentu jako takie jest w eatApple))